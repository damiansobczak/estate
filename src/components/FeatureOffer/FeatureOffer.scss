@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.featureOffer {
  padding: 90px 0;
  position: relative;

  &:before {
    content: attr(data-placeholder);
    position: absolute;
    bottom: -20px;
    left: 0;
    right: 0;
    color: $color-neutral-200;
    font-size: 180px;
    text-align: center;
    font-weight: 700;
    text-transform: uppercase;

    @include breakpoint-down(lg) {
      font-size: 100px;
      bottom: 0;
    }

    @include breakpoint-down(sm) {
      font-size: 60px;
    }
  }

  &__link {
    color: $color-accent;
    font-family: poppins, sans-serif;
    font-weight: 600;
    font-style: normal;
  }

  &__title {
    color: $color-text-primary;
  }

  &__contact {
    text-align: right;
    color: $color-text-secondary;
    text-transform: capitalize;
    font-weight: 600;

    @include breakpoint-down(md) {
      text-align: left;
    }
  }

  &__phone {
    font-weight: 700;
    color: $color-text-primary;
  }

  &__counter {
    background-color: $color-neutral-900;
    padding: 35px;
    border-radius: 6px;
    position: absolute;
    top: 40%;
    transform: translateY(-50%);
    left: -5%;
    z-index: 10;
    text-align: center;
    box-shadow: 0 30px 60px rgba(0, 0, 0, 0.3);

    @include breakpoint-down(lg) {
      display: none;
    }

    &__digit {
      color: #fff;
      font-size: $font-600;
      font-family: poppins, sans-serif;
      font-weight: 600;
      padding: 15px 0;
      margin: 5px 0 20px;
      position: relative;

      &:before {
        content: "";
        position: absolute;
        left: 0;
        right: 0;
        top: 0;
        height: 1px;
        width: 60%;
        margin: 0 auto;
        background-color: rgba($color-neutral-400, 0.5);
      }

      &:after {
        content: "";
        position: absolute;
        left: 0;
        right: 0;
        bottom: 0;
        height: 1px;
        width: 60%;
        margin: 0 auto;
        background-color: rgba($color-neutral-400, 0.5);
      }
    }

    &__text {
      color: $color-accent;
      font-family: poppins, sans-serif;
      font-weight: 600;
      font-size: $font-200;
    }
  }

  .column {
    position: relative;
  }
}
