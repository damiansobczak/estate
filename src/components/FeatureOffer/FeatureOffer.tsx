import React from "react";
import { FeatureOfferProps } from "./Interfaces/FeatureOfferProps";
import "./FeatureOffer.scss";
import Offer from "../Offer/Offer";
import Image from "../../assets/feature-offer.png";

class FeatureOffer extends React.Component<FeatureOfferProps> {
  render() {
    return (
      <section className="featureOffer" data-placeholder={this.props.placeholder}>
        <div className="container">
          <div className="featureOffer__counter">
            <div className="featureOffer__counter__digit">{this.props.count}</div>
            <div className="featureOffer__counter__text">Aktualnie w ofercie</div>
          </div>
          <div className="columns">
            <div className="column is-6-tablet">
              <div className="featureOffer__link">Zobacz więcej ofert</div>
              <h2 className="featureOffer__title">{this.props.title}</h2>
            </div>
            <div className="column is-6-tablet">
              <div className="featureOffer__contact">
                Potrzebujesz pomocy agenta? <span className="featureOffer__phone">+48 793 469 554</span>
              </div>
            </div>
          </div>
          <div className="columns">
            {this.props.data && this.props.data.slice(0, this.props.show).map((item: any) => <Offer id={item.id} image={Image} price={item.acf.cena} category={this.props.category} date="24 Czer, 2019" title={`${item.title.rendered}, ${item.acf.ulica}`} tags={[`Liczba pokoi ${item.acf.liczba_pokoi}`, `${item.acf.metraz}`, `Liczba pięter ${item.acf.liczba_pieter}`]} key={item.id}/>)}
          </div>
        </div>
      </section>
    );
  }
}

export default FeatureOffer;
