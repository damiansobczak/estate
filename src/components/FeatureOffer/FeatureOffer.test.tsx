import React from "react";
import { render } from "enzyme";
import renderer from "react-test-renderer";
import FeatureOffer from "./FeatureOffer";
import { BrowserRouter } from "react-router-dom";

const data = {
    "title": "Oferty najmu nieruchomości",
    "placeholder": "Wynajem",
    "category": "Wynajem",
    "show": 3,
    "data": [
        {
            "id": 45,
            "slug": "osiedle-bialy-las",
            "link": "http://localhost:8888/estate/nieruchomosc/osiedle-bialy-las/",
            "title": {
                "rendered": "Osiedle Biały Las"
            },
            "content": {
                "rendered": "\n<p>Przedstawiamy najnowszą ofertę biura nieruchomości. Inwestycja usytuowana w ustronnym zaciszu, z dala od miejskiego zgiełku. Poniżej znajdą Państwo wszelkie niezbędne informacje odnośnie oferty w tym: miejsce usytuowania, dodatkowe wyposażenie oraz całkowite koszty mieszkalne.<br></p>\n",
                "protected": false
            },
            "excerpt": {
                "rendered": "<p>Przedstawiamy najnowszą ofertę biura nieruchomości. Inwestycja usytuowana w ustronnym zaciszu, z dala od miejskiego zgiełku. Poniżej znajdą Państwo wszelkie niezbędne informacje odnośnie oferty w tym: miejsce usytuowania, dodatkowe wyposażenie oraz całkowite koszty mieszkalne.</p>\n",
                "protected": false
            },
            "featured_media": 79,
            "categories": [
                2
            ],
            "better_featured_image": {
                "id": 79,
                "source_url": "http://localhost:8888/estate/wp-content/uploads/2020/01/image-5.jpg"
            },
            "acf": {
                "ulica": "ul. Leśna 2",
                "metraz": "120m2",
                "liczba_pokoi": "5",
                "cena": "345,700zł",
                "typ_domu": "Dom jednorodzinny",
                "liczba_pieter": "2",
                "miejsce_parkingowe": "2",
                "powierzchnia_dzialki": "35m2",
                "promocja": true,
                "wspolrzedne": {
                    "lat": "51.748378",
                    "lng": "19.464890"
                },
                "kaucja": "2000",
                "media": "230",
                "czynsz": "450",
                "atrybuty": {
                    "lazienka": "Prysznic, pralka",
                    "pokoje": "2, TV, stół, sofa",
                    "kuchnia": "Kuchenka, czajnik",
                    "balkon": "6m2, zastawa",
                    "parking": "Miejsce, szlaban",
                    "metraz": "26m2, taras"
                },
                "tlo": false,
                "galeria": {
                    "obraz_galerii_1": "http://localhost:8888/estate/wp-content/uploads/2020/01/image-6.jpg",
                    "obraz_galerii_2": "http://localhost:8888/estate/wp-content/uploads/2020/01/image-7.jpg",
                    "obraz_galerii_3": "http://localhost:8888/estate/wp-content/uploads/2020/01/image-3.jpg",
                    "obraz_galerii_4": "http://localhost:8888/estate/wp-content/uploads/2020/01/image-2.jpg"
                }
            }
        }
    ]
}

describe("<FeatureOffer />", () => {
    it("renders component correctly", () => {
        const tree = renderer.create(
            <BrowserRouter>
                <FeatureOffer
                    title="Oferty sprzedaży nieruchomości"
                    placeholder="Sprzedaż"
                    category="Sprzedaż"
                    show={3}
                    data={data.data} />
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("show total offers numer", () => {
        const wrapper = render(
            <BrowserRouter>
                <FeatureOffer
                    title="Oferty sprzedaży nieruchomości"
                    placeholder="Sprzedaż"
                    category="Sprzedaż"
                    show={3}
                    data={data.data} />
            </BrowserRouter>
        );
        expect(wrapper.find(".featureOffer__counter__digit").html()).toContain("1");
    });

    it("display offer correctly", () => {
        const wrapper = render(
            <BrowserRouter>
                <FeatureOffer
                    title="Oferty sprzedaży nieruchomości"
                    placeholder="Sprzedaż"
                    category="Sprzedaż"
                    show={3}
                    data={data.data} />
            </BrowserRouter>
        );
        expect(wrapper.find(".offer__media > img").prop("src")).toContain("http://localhost:8888/estate/wp-content/uploads/2020/01/image-5.jpg");
        expect(wrapper.find(".offer__title").html()).toContain("Osiedle Bia&#x142;y Las, ul. Le&#x15B;na 2");
    });

    it("display props correctly", () => {
        const wrapper = render(
            <BrowserRouter>
                <FeatureOffer
                    title="Oferty sprzedaży nieruchomości"
                    placeholder="Sprzedaż"
                    category="Sprzedaż"
                    show={3}
                    data={data.data} />
            </BrowserRouter>
        );
        expect(wrapper.find(".featureOffer__title").html()).toContain("Oferty sprzeda&#x17C;y nieruchomo&#x15B;ci");
    });
});
