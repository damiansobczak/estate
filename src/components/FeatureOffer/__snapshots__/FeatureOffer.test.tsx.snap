// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeatureOffer /> renders component correctly 1`] = `
<section
  className="featureOffer"
  data-placeholder="Sprzedaż"
>
  <div
    className="container"
  >
    <div
      className="featureOffer__counter"
    >
      <div
        className="featureOffer__counter__digit"
      >
        1
      </div>
      <div
        className="featureOffer__counter__text"
      >
        Aktualnie w ofercie
      </div>
    </div>
    <div
      className="columns"
    >
      <div
        className="column is-6-tablet"
      >
        <a
          className="featureOffer__link"
          href="/sprzedaz"
          onClick={[Function]}
        >
          Zobacz więcej ofert
        </a>
        <h2
          className="featureOffer__title"
        >
          Oferty sprzedaży nieruchomości
        </h2>
      </div>
      <div
        className="column is-6-tablet"
      >
        <div
          className="featureOffer__contact"
        >
          Potrzebujesz pomocy agenta? 
          <span
            className="featureOffer__phone"
          >
            +48 793 469 554
          </span>
        </div>
      </div>
    </div>
    <div
      className="columns is-multiline"
    >
      <a
        className="column is-6-tablet is-4-desktop"
        href="/oferta/45"
        onClick={[Function]}
      >
        <div
          className="offer__favorite "
          onClick={[Function]}
        >
          <span
            className="icon-heart"
          />
        </div>
        <div
          className="offer"
        >
          <div
            className="offer__media"
          >
            <img
              alt=""
              src="http://localhost:8888/estate/wp-content/uploads/2020/01/image-5.jpg"
            />
            <div
              className="offer__price"
            >
              345,700zł
            </div>
          </div>
          <div
            className="offer__content"
          >
            <div
              className="offer__meta"
            >
              <div
                className="offer__category"
              >
                Sprzedaż
              </div>
              <div
                className="offer__date"
              >
                24 Czer, 2019
              </div>
            </div>
            <div
              className="offer__title"
            >
              Osiedle Biały Las, ul. Leśna 2
            </div>
            <div
              className="offer__sep"
            />
            <div
              className="offer__tags"
            >
              Liczba pokoi 5, 
              120m2, 
              Liczba pięter 2, 
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</section>
`;
